smart-send-logistics/build/smart-send-block-frontend.js

```diff
@@ -0,0 +1,667 @@
+  const [hasInteractedWithOtherInput, setHasInteractedWithOtherInput] = (0,_wordpress_element__WEBPACK_IMPORTED_MODULE_1__.useState)(false);
+  (0,_wordpress_element__WEBPACK_IMPORTED_MODULE_1__.useEffect)(() => {
+    if (setselectedPickupPoint !== "other") {
+      if (validationError) {
+        clearValidationError(validationErrorId);
+      }
+      return;
+    }
+    setValidationErrors({
+      [validationErrorId]: {
+        message: (0,_wordpress_i18n__WEBPACK_IMPORTED_MODULE_3__.__)("Please add some text", "smart-send-logistics"),
+        hidden: !hasInteractedWithOtherInput
+      }
+    });
+  }, [clearValidationError, setselectedPickupPoint, setValidationErrors, validationErrorId, debouncedSetExtensionData, validationError]);
+  return (0,react__WEBPACK_IMPORTED_MODULE_0__.createElement)("div", {
+    className: "wp-block-smart-send-not-at-home"
+  }, (0,react__WEBPACK_IMPORTED_MODULE_0__.createElement)(_wordpress_components__WEBPACK_IMPORTED_MODULE_2__.SelectControl, {
+    label: (0,_wordpress_i18n__WEBPACK_IMPORTED_MODULE_3__.__)("TLS Delivery Point", "smart-send-logistics"),
+    value: selectedPickupPoint,
+    options: _options__WEBPACK_IMPORTED_MODULE_6__.options,
+    onChange: setselectedPickupPoint,
+    className: "shiiping_smart_ar_hide"
+  }));
+};
+
+// Function to find closest agent by address
+async function findClosestAgentByAddress(ss_agent, country, postalCode, city, street) {
+  country = country;
+  postalCode = postalCode;
+  city = city;
+  street = street;
+  var carrier = await getShippingCarrier(ss_agent);
+  getPickupPoints(carrier, country, postalCode, city, street);
+}
+function setValue(id) {
+  var element = document.getElementById(id);
+  if (element) {
+    return element.value;
+  }
+  return null;
+}
+const getPickupPoints = async (carrier, country, postalCode, city, street) => {
+  const url = "/wp-json/smart-send-logistics/v1/get-closest-pickup-points"; // Your endpoint URL
+
+  const data = {
+    country: country,
+    postCode: postalCode,
+    city: city,
+    street: street,
+    carrier: carrier
+  };
+  try {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(data)
+    });
+    if (!response.ok) {
+      const errorData = await response.json();
+      jQuery(".shiiping_smart_ar_hide").hide();
+      jQuery(".shiiping_smart_ar_hide").css("opacity", 0);
+    } else {
+      const results = await response.json();
+      const sortedData = Object.entries(results).sort((a, b) => {
+        const distanceA = a[0] === "0" ? 0 : parseFloat(a[1].split(":")[0].replace("km", "").replace("m", "")) * (a[1].includes("km") ? 1000 : 1);
+        const distanceB = b[0] === "0" ? 0 : parseFloat(b[1].split(":")[0].replace("km", "").replace("m", "")) * (b[1].includes("km") ? 1000 : 1);
+        return distanceB - distanceA;
+      });
+      var reversedrestoreddata = sortedData.reverse();
```

Comment:
Frontend should do no sorting. Sorting is part of the API so results should be shown in exactly the same order as returned by the API.

Reply:
We are currently sorting the results to retain the original order as received from the API response. During the process of converting these results into JSON format, the alignment can be inadvertently altered, particularly when numeric keys are involved. This disruption can affect the consistency of the data presentation. To address this, we are re-sorting the data to ensure it adheres to the initial sequence provided by the API.

However, if we want to avoid this re-sorting step, one approach would be to bypass the JSON conversion entirely. This, though, is not considered an ideal solution due to potential limitations in data handling and compatibility.

An alternative solution would be to modify the API response structure, delivering the data as a simple nested array rather than an associative array. This adjustment would allow the front-end to process the data without requiring additional sorting, thereby preserving the original order directly from the API.

**Important Link**: https://stackoverflow.com/questions/42420916/prevent-javascript-from-changing-the-order-of-the-json

**Please advise on which solution you would prefer us to implement.**